### PR TITLE
Closes #5965: Migrate concept-tabstray and browser-tabstray to use browser-state.

### LINKED ISSUE #5965: Migrate browser-tabstray to use browser-state

### PR DESCRIPTION
**Pull request by pocmo**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [x] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by pocmo** [comment]
Need to update the `feature-tabs` tests.

**Comment by pocmo** [comment]
Okay, this is now ready to land.

File: components/concept/tabstray/src/main/java/mozilla/components/concept/tabstray/Tab.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.tabstray
+
+import android.graphics.Bitmap
+
+/**
+ * Data class representing a tab to be displayed in a [TabsTray].
+ *
+ * @param id Unique ID of the tab.
```
Comment (csadilek) [inline components/concept/tabstray/src/main/java/mozilla/components/concept/tabstray/Tab.kt]: nit: `@property`?

File: components/concept/tabstray/src/main/java/mozilla/components/concept/tabstray/Tabs.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.tabstray
+
+/**
+ * Aggregate data type keeping a reference to the list of tabs and the index of the selected tab.
+ *
+ * @param list The list of tabs.
```
Comment (csadilek) [inline components/concept/tabstray/src/main/java/mozilla/components/concept/tabstray/Tabs.kt]: nit: `@property`

**Comment by pocmo** [comment]
Updated the PR. Will hold back landing to synchronize snapshots & Fenix PR on Monday morning.

**Comment by pocmo** [comment]
Updated changelog and prepared PR for Fenix.

bors r=csadilek

**Pull request merged**